syntax = "proto3";

package pet;

import "google/protobuf/empty.proto";

// Service for pet adoption
service PetAdoptionService {
    rpc RegisterPet(Pet) returns (Response);
    rpc SearchPet(SearchRequest) returns (SearchResponse);
}

// Message to register a new pet
message Pet {
    string name = 1;
    string gender = 2;
    int32 age = 3;
    string breed = 4;
    string picture_url = 5;  // URL to the pet's picture
}

// Message for the response after registering
message Response {
    string message = 1;
}

// Message to search for a pet
message SearchRequest {
    string query = 1;  // Name, gender, age, or breed to search
}

// Response containing a list of matching pets
message SearchResponse {
    repeated Pet pets = 1;  // List of matching pets
}
